src/middlewares/Cache.spec.ts

```diff
@@ -0,0 +1,117 @@
+import {Cache, ICacheMiddlewareStore} from "./Cache";
+import {FetchResponse} from "./Fetch";
+
+declare const Response: FetchResponse<string>;
+declare const localStorage: ICacheMiddlewareStore;
+
+describe("Cache", () => {
+    it("is defined", () => {
+        expect(Cache).toBeDefined();
+    });
+
+    describe("constructor()", () => {
+        it("can be instantiated", () => {
+            expect(new Cache() instanceof Cache).toBeTruthy();
+        });
+
+        it("accepts storage engine in instantiation", () => {
+            expect(new Cache(localStorage) instanceof Cache).toBeTruthy();
+        });
+    });
+
+    describe("process()", () => {
+        it("is a function", () => {
+            expect(typeof (new Cache()).process).toBe("function");
+        });
+
+        it("calls next() if cache not enabled", () => {
+            const cached = new Cache().process({}, () => "just any thing" as any);
+
+            expect(cached).toBe("just any thing");
+        });
+
+        it("calls next() if cache expired", (done) => {
+            const cache = { enable: true, maxAge: 10 }; // 10 seconds
+            const mock: Promise<any> = Promise.resolve(new Response("fetched response", undefined));
+
+            // Seed the cache with expired timestamp for test
+            localStorage.setItem(
+                "/test",
+                JSON.stringify({
+                    value: "cached response",
+                    timestamp: +Date.now() - 12000, // 12 seconds before
+                    headers: { "Content-Type": "text/html" }
+                })
+            );
+
+            const cached = new Cache(localStorage).process({ url: "/test", cache }, () => done() && 0 || mock);
```

Comment:
to ensure that `next()` cb has been called for some cases. the test expects `done()` be called for passing. here `done()` has been called only when `next` is invoked thus proving that test passes when `next()` is fired.